Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -457,6 +464,24 @@ public final class PackageBuilder {
     /// Private function that creates and returns a list of targets defined by a package.
     private func constructTargets() throws -> [Target] {
 
+        // Ensure no dupicate target definitions are found
+        // Find duplicate targets
+        var uniqueTargets = Set<String>()
+        var duplicateTargets  = Array<String>()
+        for target in manifest.package.targets {
+            guard !uniqueTargets.contains(target.name) else {
+                duplicateTargets.append(target.name)
+                continue
+            }
+
+            uniqueTargets.insert(target.name)
+        }
```

Comment:
Looks like we're checking for duplicates in graph [here](https://github.com/apple/swift-package-manager/blob/master/Sources/PackageGraph/PackageGraphLoader.swift#L171), but that logic feels broken if two indirectly dependent packages have same target names. I filed a separate bug for that: https://bugs.swift.org/browse/SR-4859